Source/Models/Highlight.swift

```diff
@@ -7,139 +7,129 @@
-            completion?(error: nil)
-        } catch let error as NSError {
-            completion?(error: error)
-        }
+    dynamic var bookId:String!
+    dynamic var content:String!
+    dynamic var contentPost:String!
+    dynamic var contentPre:String!
+    dynamic var date:NSDate!
+    dynamic var highlightId:String!
+    dynamic var page:Int = 0
+    dynamic var type:Int = 0
+    dynamic var startOffset:Int = -1
+    dynamic var endOffset:Int = -1
+    
+    override class func primaryKey()-> String{
+        return "highlightId"
     }
     
     public static func persistHighlight(object: FRHighlight, completion: Completion?) {
-        var highlight: Highlight?
-        
         do {
-            let fetchRequest = NSFetchRequest(entityName: "Highlight")
-            fetchRequest.predicate = NSPredicate(format:"highlightId = %@", object.id)
-            highlight = try coreDataManager.managedObjectContext.executeFetchRequest(fetchRequest).last as? Highlight
-        } catch let error as NSError {
-            print(error)
-            highlight = nil
-        }
-  
-        if highlight != nil {
-            highlight!.content = object.content
-            highlight!.contentPre = object.contentPre
-            highlight!.contentPost = object.contentPost
-            highlight!.date = object.date
-            highlight!.type = object.type.hashValue
-        } else {
-            highlight = NSEntityDescription.insertNewObjectForEntityForName("Highlight", inManagedObjectContext: coreDataManager.managedObjectContext) as? Highlight
-            coreDataManager.saveContext()
-
-            highlight!.bookId = object.bookId
-            highlight!.content = object.content
-            highlight!.contentPre = object.contentPre
-            highlight!.contentPost = object.contentPost
-            highlight!.date = NSDate()
-            highlight!.highlightId = object.id
-            highlight!.page = object.page
-            highlight!.type = object.type.hashValue
-        }
-
-        // Save
-        do {
-            try coreDataManager.managedObjectContext.save()
-            completion?(error: nil)
+            let realm = try! Realm()
+            
+            let newHighlight = Highlight()
+            newHighlight.bookId = object.bookId
+            newHighlight.content = object.content
+            newHighlight.contentPre = object.contentPre
+            newHighlight.contentPost = object.contentPost
+            newHighlight.date = NSDate()
+            newHighlight.highlightId = object.id
+            newHighlight.page = object.page
+            newHighlight.type = object.type.hashValue
+            newHighlight.startOffset = object.startOffset
+            newHighlight.endOffset = object.endOffset
+            
+            realm.beginWrite()
+            realm.add(newHighlight, update: true)
+            try! realm.commitWrite()
         } catch let error as NSError {
-            completion?(error: error)
```

Comment:
I think we need to call the completion here, because the function is waiting for that.